psi4/src/psi4/dfmp2/mp2.cc

```diff
@@ -2411,16 +2418,18 @@ void RDFMP2::form_Z()
         Dtemp->scale(0.5);
 
         // Add in the reference contribution
+
         for (int i = 0; i < nocc; ++i)
             Dtemp->add(i, i, 1.0);
-        Ca_ = SharedMatrix(new Matrix("DF-MP2 (unrelaxed) Natural Orbitals", nsopi_, nmopi_));
-        epsilon_a_ = SharedVector(new Vector("DF-MP2 (unrelaxed) NO Occupations", nmopi_));
+        Ca__ = SharedMatrix(new Matrix("DF-MP2 (unrelaxed) Natural Orbitals", nsopi_, nmopi_));
+        epsilon_a__ = SharedVector(new Vector("DF-MP2 (unrelaxed) NO Occupations", nmopi_));
```

Comment:
As you pointed out, many of the post-scf methods don't set their energy_ and just update the process environment variable. This needs a major overhaul. Here, I just wanted to fix the problem faced by a user on the forum. Also, I was uncertain if we needed the unrelaxed natural orbitals coefficients and occupation numbers as I didn't see them getting used anywhere. How do you suggest I retain these as I want the Ca_ and epsilon_a_ to be the conventional MO coefficients and orbital energies respectively ?